File: scratch/analyze_assets.js
const fs = require('fs');
const path = require('path');

const srcDir = "c:\\Users\\miaje\\OneDrive\\Desktop\\EncuentroMundialDeValores\\src";
const speakersDir = "c:\\Users\\miaje\\OneDrive\\Desktop\\EncuentroMundialDeValores\\public\\speakers";
const sponsorsDir = "c:\\Users\\miaje\\OneDrive\\Desktop\\EncuentroMundialDeValores\\public\\Sponsors";
const imagesDir = "c:\\Users\\miaje\\OneDrive\\Desktop\\EncuentroMundialDeValores\\public\\images";
const indexHtmlPath = "c:\\Users\\miaje\\OneDrive\\Desktop\\EncuentroMundialDeValores\\index.html";

// Read all files recursively in srcDir and append contents
let srcContent = "";

function readFilesRecursively(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            readFilesRecursively(fullPath);
        } else if (file.endsWith('.jsx') || file.endsWith('.js') || file.endsWith('.css') || file.endsWith('.html')) {
            srcContent += fs.readFileSync(fullPath, 'utf8') + "\n";
        }
    }
}

if (fs.existsSync(srcDir)) {
    readFilesRecursively(srcDir);
}

if (fs.existsSync(indexHtmlPath)) {
    srcContent += fs.readFileSync(indexHtmlPath, 'utf8') + "\n";
}

function checkUnused(directory, name) {
    console.log(`\n--- Checking unused files in ${name} ---`);
    if (!fs.existsSync(directory)) {
        console.log(`Directory ${directory} does not exist.`);
        return [];
    }

    const unused = [];
    const files = fs.readdirSync(directory);
    for (const file of files) {
        if (file === ".DS_Store") continue;
        
        // Exact matching or finding substring in the source code
        if (!srcContent.includes(file)) {
            unused.push(file);
            console.log(`Unused: ${file}`);
        } else {
            console.log(`Used: ${file}`);
        }
    }
    return unused;
}

const unusedSpeakers = checkUnused(speakersDir, "speakers");
const unusedSponsors = checkUnused(sponsorsDir, "Sponsors");
const unusedImages = checkUnused(imagesDir, "images");

console.log("\nSummary of unused files:");
console.log(`Speakers unused count: ${unusedSpeakers.length}`);
console.log(`Sponsors unused count: ${unusedSponsors.length}`);
console.log(`Images unused count: ${unusedImages.length}`);

File: scratch/analyze_assets.py
import os
import re

src_dir = r"c:\Users\miaje\OneDrive\Desktop\EncuentroMundialDeValores\src"
speakers_dir = r"c:\Users\miaje\OneDrive\Desktop\EncuentroMundialDeValores\public\speakers"
sponsors_dir = r"c:\Users\miaje\OneDrive\Desktop\EncuentroMundialDeValores\public\Sponsors"
images_dir = r"c:\Users\miaje\OneDrive\Desktop\EncuentroMundialDeValores\public\images"

# Gather all text content from src files
src_content = ""
for root, dirs, files in os.walk(src_dir):
    for f in files:
        if f.endswith(('.jsx', '.js', '.css', '.html')):
            path = os.path.join(root, f)
            with open(path, 'r', encoding='utf-8', errors='ignore') as file_obj:
                src_content += file_obj.read() + "\n"

# Also include index.html
with open(r"c:\Users\miaje\OneDrive\Desktop\EncuentroMundialDeValores\index.html", 'r', encoding='utf-8', errors='ignore') as f:
    src_content += f.read()

def check_unused(directory, name):
    print(f"\n--- Checking unused files in {name} ---")
    if not os.path.exists(directory):
        print(f"Directory {directory} does not exist.")
        return []
    
    unused = []
    files = os.listdir(directory)
    for f in files:
        if f == ".DS_Store":
            continue
        # Check if the filename (or part of it) is in the source code
        # We can look for exact match of the filename
        # Since sometimes paths are like "/speakers/name.jpg" or "name.jpg"
        if f not in src_content:
            # Let's also do a check ignoring casing/quotes just in case
            unused.append(f)
            print(f"Unused: {f}")
        else:
            print(f"Used: {f}")
    return unused

unused_speakers = check_unused(speakers_dir, "speakers")
unused_sponsors = check_unused(sponsors_dir, "Sponsors")
unused_images = check_unused(images_dir, "images")

print("\nSummary of unused files:")
print(f"Speakers unused count: {len(unused_speakers)}")
print(f"Sponsors unused count: {len(unused_sponsors)}")
print(f"Images unused count: {len(unused_images)}")
